apis/python/src/tiledbsc/soma.py

```diff
@@ -547,6 +547,98 @@ def from_soma_slice(
 
         return soma
 
+    @classmethod
+    def map(
+        cls,
+        somas: List[SOMA],
+        soma_callback: Callable[[SOMA, Optional[Any]], Any],
```

Comment:
If somewhere at the module level you add

```python
_T = TypeVar("_T")
```

then here you can change `soma_callback` to be a `Callable[[Soma, Optional[Any]], _T]`, and make the return type `Dict[str, _T]`. That will give the caller an even better type-checking experience, since now the return type from the callable will be automatically carried through to the result.

If you do this, you’ll also want to make the same change down in the `_map_...` functions.

(note: suggesting the use of `_T` rather than `T` as a name so that the TypeVar variable doesn’t look exported. using just `T` is semantically exactly the same and completely fine if you think it’s better)